src/Cloud.Core/IMessenger.cs

```diff
@@ -48,6 +48,14 @@ public interface IMessageOperations : INamedInstance
         /// <returns>Dictionary of string, object properties.</returns>
         IDictionary<string, object> ReadProperties<T>(T msg) where T : class;
 
+        /// <summary>
+        /// Read system properties from a message.
+        /// </summary>
+        /// <typeparam name="T">Type T of message.</typeparam>
+        /// <param name="msg">Message body, used to identity the message to read from.</param>
+        /// <returns>Dictionary of string, object system properties.</returns>
+        IDictionary<string, object> ReadSystemProperties<T>(T msg) where T : class;
```

Comment:
I wonder would it be better to just use the read properties and included system props within that?

Reply:
Yes that sounds good to me, it would be useful to have them available in IMessageEntity rather than having to specifically read too.